sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/OpenCoverReportParserTest.java

```diff
@@ -108,6 +108,29 @@ public void valid() throws Exception {
     assertThat(debugLogs.get(0)).startsWith("The current user dir is '");
   }
 
+  @Test
+  public void branchCoverage() throws Exception {
+    Coverage coverage = new Coverage();
+    String filePath = new File("MyLibrary\\Calculator.cs").getCanonicalPath();
+
+    new OpenCoverReportParser(alwaysTrue).accept(new File("src/test/resources/opencover/coverage_branches.xml"), coverage);
+
+    assertThat(coverage.files()).containsOnly(filePath);
+    assertThat(coverage.hits(filePath))
+      .hasSize(3)
+      .contains(
+        Assertions.entry(6, 1),
+        Assertions.entry(7, 1),
+        Assertions.entry(14, 1));
+
+    assertThat(coverage.getLinesBranchCoverage(filePath).size()).isEqualTo(1);
```

Comment:
could you also check the branch coverage at line 2433 in the test XML?
- you'll be able to verify that all get parsed
- you'll be able to verify the first, middle and last entry to make sure they get read correctly